extra_foam/utils.py

```diff
@@ -42,6 +42,31 @@ def timed_f(*args, **kwargs):
     return wrap
 
 
+
+class BlockTimer:
+    """A context manager for measuring the execution time of a code block
+
+    For example::
+
+        >>> with BlockTimer("foo"):
+        ...     time.sleep(1)
+        ...
+        Execution of foo: 1.001s
+    """
+    def __init__(self, label="block", enabled=True):
+        self._label = label
+        self._enabled = enabled
+
+    def __enter__(self):
+        self.start = time.perf_counter()
+        return self
+
+    def __exit__(self, *_):
```

Comment:
nitpicking: we usually use `*args` and `**kwargs` for arguments and keyword arguments in python.
You could also use the positional arguments, even if you don't use them, ` exception_type, exception_value, traceback`

Reply:
Hmm, I don't really like that style, to me it's a bit weird to take named arguments if they're never used. If they were used then I would indeed have used `*args` or `exeption_type, ...`.

Reply:
as said, it's just nitpicking about the common practice in Python, not forcing you to change it